### ISSUE TITLE
default device on transformers notebook

### ISSUE THREAD

**Issue by EdoardoPona** [EffiSciencesResearch/ML4G-2.0#1]
The reference transformer is not explicitly moved to the device, this causes issues when the default device does not coincide with 'cpu' (ie. on Mac)

https://github.com/EffiSciencesResearch/ML4G-2.0/blob/master/workshops/transformer/transformer.ipynb

**Comment by EdoardoPona** [comment]
the reference transformer which is initialised like so:

```python

device = torch.device("cpu")

reference_gpt2 = HookedTransformer.from_pretrained(
    "gpt2-small", fold_ln=False, center_unembed=False, center_writing_weights=False
)
```

should be explicitly moved to the device like so

```python
reference_gpt2 = HookedTransformer.from_pretrained(
    "gpt2-small", fold_ln=False, center_unembed=False, center_writing_weights=False
).to(device)
```

This in all 3 of the transformer notebooks: 
https://github.com/EffiSciencesResearch/ML4G-2.0/blob/master/workshops/transformer/transformer.ipynb
https://github.com/EffiSciencesResearch/ML4G-2.0/blob/master/workshops/transformer/transformer_hard.ipynb
https://github.com/EffiSciencesResearch/ML4G-2.0/blob/master/workshops/transformer/transformer_normal.ipynb

**Comment by ddorn** [comment]
This was closed in 1506f95.
